include/picolibrary/circular_buffer.h

```diff
@@ -24,6 +24,77 @@
 #define PICOLIBRARY_CIRCULAR_BUFFER_H
 
 namespace picolibrary {
+
+/**
+ * \brief Circular buffer use case.
+ *
+ * \attention These use cases assume that interrupts that interact with the circular
+ *            buffer cannot be interrupted by other interrupts that interact with the same
+ *            circular buffer.
+ */
+enum class Circular_Buffer_Use_Case {
+    /**
+     * The main thread of execution reads from and writes to the circular buffer.
+     * Interrupts do not interact with the circular buffer.
+     */
+    MAIN_READS_WRITES,
+
+    /**
+     * An interrupt reads from and writes to the circular buffer. Other interrupts do not
+     * interact with the circular buffer. The main thread of execution does not interact
+     * with the circular buffer.
+     */
+    INTERRUPT_READS_WRITES,
+
+    /**
+     * An interrupt reads from the circular buffer. One or more other interrupts write to
+     * the circular buffer. The main thread of execution does not interact with the
+     * circular buffer.
+     */
+    INTERRUPT_READS_INTERRUPT_WRITES,
+
```

Comment:
Add `INTERRUPT_READS_WRITES_INTERRUPT_WRITES` as an alias for `INTERRUPT_READS_INTERRUPT_WRITES` since the implementations for both will be identical.